src/uharfbuzz/_harfbuzz.pyx

```diff
@@ -1369,3 +1401,375 @@ def repack_with_tag(tag: str,
+        hb_tag = hb_tag_from_string(tag.encode("ascii"), -1)
+        return hb_subset_input_pin_axis_to_default(
+            self._input, face._hb_face, hb_tag
+        )
+
+    def pin_axis_location(self, face: Face, tag: str, value: float) -> bool:
+        hb_tag = hb_tag_from_string(tag.encode("ascii"), -1)
+        return hb_subset_input_pin_axis_location(
+            self._input, face._hb_face, hb_tag, value
+        )
+
+    @property
+    def unicode_set(self) -> Set:
+        return Set.from_ptr(hb_set_reference (hb_subset_input_unicode_set(self._input)))
+
+    @property
+    def glyph_set(self) -> Set:
+        return Set.from_ptr(hb_set_reference (hb_subset_input_glyph_set(self._input)))
+
+    def sets(self, set_type : SubsetInputSets) -> Set:
+        return Set.from_ptr(hb_set_reference (hb_subset_input_set(self._input, set_type)))
+
+    def __getitem__(self, set_type : SubsetInputSets) -> Set:
+        return self.sets(set_type)
+
+    def __setitem__(self, set_type : SubsetInputSets, iterable) -> Set:
+        s = self.sets(set_type)
+        s.set(iterable)
+
+    @property
+    def flags(self) -> SubsetFlags:
+        cdef unsigned subset_flags = hb_subset_input_get_flags(self._input)
+        return SubsetFlags(subset_flags)
+
+    @flags.setter
+    def flags(self, flags: SubsetFlags) -> None:
+        hb_subset_input_set_flags(self._input, int(flags))
+
+
+cdef class Set:
+    cdef hb_set_t* _hb_set
+
+    INVALID_VALUE = HB_SET_VALUE_INVALID
+
+    def __cinit__(self, init = set()):
+        self._hb_set = hb_set_create()
+        if not hb_set_allocation_successful(self._hb_set):
+            raise MemoryError()
+
+        self.set(init)
+
+    def __dealloc__(self):
+        hb_set_destroy(self._hb_set)
+
+    @staticmethod
+    cdef Set from_ptr(hb_set_t* hb_set):
+        """Create Set from a pointer taking ownership of a it."""
+
+        cdef Set wrapper = Set.__new__(Set)
+        wrapper._hb_set = hb_set
+        return wrapper
+
+    def copy(self) -> Set:
+        c = Set()
+        c._hb_set = hb_set_copy(self._hb_set)
+        return c
+
+    def clear(self):
+        hb_set_clear(self._hb_set)
+
+    def is_empty(self) -> bool:
+        return hb_set_is_empty(self._hb_set)
+
+    def __bool__(self) -> bool:
+        return not self.is_empty()
+
+    def invert(self):
+        hb_set_invert(self._hb_set)
+
+    def has(self, c: int) -> bool:
+        return hb_set_has(self._hb_set, c)
+
+    def __contains__(self, c) -> bool:
+        if type(c) != int: # TODO Small-int?
+            return False
+        return self.has(c)
+
+    def add(self, c: int):
+        hb_set_add(self._hb_set, c)
+        if not hb_set_allocation_successful(self._hb_set):
+            raise MemoryError()
+
+    def add_range(self, first: int, last: int):
+        hb_set_add_range(self._hb_set, first, last)
+        if not hb_set_allocation_successful(self._hb_set):
+            raise MemoryError()
+
+    def remove(self, c: int):
```

Comment:
Idiomatically, I think this should be just `__delitem__`?

Reply:
set() also has remove and discard.

Reply:
Also: set doesn't have `__delitem__`